packages/kbn-esql-validation-autocomplete/src/validation/validation.ts

```diff
@@ -500,6 +483,43 @@ function validateFunction(
   return uniqBy(messages, ({ location }) => `${location.min}-${location.max}`);
 }
 
+/**
+ * We only want to report one message when any number of the elements in an array argument is of the wrong type
+ */
+function collapseWrongArgumentTypeMessages(
```

Comment:
Can you move this to `helpers.ts` file?

Probably other functions like `extractCompatibleSignaturesForFunction` should be moved there as well, to keep only `validate*` functions here.

Reply:
Done in https://github.com/elastic/kibana/pull/180625/commits/13549ff81336d362ffb63af9a421eaaf009a9c2b